user/src/main/java/org/tessell/widgets/IsRowTable.java

```diff
@@ -3,8 +3,16 @@
 import org.tessell.gwt.user.client.ui.IsWidget;
 import org.tessell.util.ListDiff.ListLike;
 
+import com.google.gwt.user.client.ui.Widget;
+
 public interface IsRowTable extends IsWidget, HasRows {
 
   ListLike<IsWidget> getRowsPanel();
 
+  /** Assumes {@code row} is a table row and returns its index */
+  int indexOfRow(final Widget row);
```

Comment:
Hehe...no... :-)

The other interface, HasRows, was fine.

I just meant you have two methods here:

* indexOfRow(Widget) -- is a real dom widget, can only be used at runtime
* indexOfRow(IsWidget) -- can be either a real dom widget (when IsWidget is implemented by the real widgets) or a test widget, so can be used at either runtime or test time.

Given indexOfRow(IsWidget) can be used in more places that indexOfRow, I believe it's sufficient to provide only the indexOfRow(IsWidget) overload, especially in the HasRows/IsRowTable interfaces (given they are explicitly meant for unit tests, for which indexOfRow(Widget) is not useful).

If you want to keep the two overloads of indexOfRow (one for IsWidget and one for Widget) on the concrete RowTable type (e.g. for a codebase that didn't case about unit testing, had only Widgets, and so wanted to call RowTable.indexOfRow(Widget) directly), that would be fine--just trying pointing out it (indexOfRow(Widet)) is useful in the concrete RowTable, but not really the test interfaces.